crates/primitives/src/proofs.rs

```diff
@@ -104,4 +126,157 @@ mod tests {
             H256(hex!("fe70ae4a136d98944951b2123859698d59ad251a381abc9960fa81cae3d0d4a0"))
         );
     }
+
+    #[test]
+    fn check_empty_state_root() {
+        let genesis_alloc = HashMap::new();
+        let root = genesis_state_root(genesis_alloc);
+        assert_eq!(root, EMPTY_ROOT);
+    }
+
+    #[test]
+    fn test_simple_account_state_root() {
+        // each fixture specifies an address and expected root hash - the address is initialized
+        // with a maximum balance, and is the only account in the state.
+        // these test cases are generated by using geth with a custom genesis.json (with a single
+        // account that has max balance)
+        let fixtures: Vec<(Address, H256)> = vec![
+            (
+                hex!("9fe4abd71ad081f091bd06dd1c16f7e92927561e").into(),
+                hex!("4b35be4231841d212ce2fa43aedbddeadd6eb7d420195664f9f0d55629db8c32").into(),
+            ),
+            (
+                hex!("c2ba9d87f8be0ade00c60d3656c1188e008fbfa2").into(),
+                hex!("e1389256c47d63df8856d7729dec9dc2dae074a7f0cbc49acad1cf7b29f7fe94").into(),
+            ),
+        ];
```

Comment:
how were these chosen? do we know this is the right state root?

Reply:
These were chosen by doing the following:
 - Generating an address
 - Creating a `genesis.json` giving each addresses `U256::max()` coins
 - Initializing and starting geth with this `genesis.json`
 - Retrieving the genesis block over RPC
 - Getting the state root from the returned block

So these are not official test cases, and to reproduce these test cases the above procedure would have to be repeated or scripted in a test.